### ISSUE TITLE
:on-event-create does not work

### ISSUE THREAD

**Issue by vip9** [antoniandre/vue-cal-v4#299]
When using :on-event-create with Click&Hold the function is not fired.
Affected version 3.5.1
3.4.1 is fine

https://codepen.io/vipw/pen/zYrBByO

**Comment by giulianimartini** [comment]
Hey @vip9,

I just added `:drag-to-create-event="false"` and then it started to work.

Do you need the `drag-to-create-event` property to be true?

Hope it helps!

**Comment by vip9** [comment]
Thanks, I do not use drag to create for now. The trick did the job.
But found out (even on ver 5.4.2), that when created event, there's no "split" property in selectedEvent object.

**Comment by giulianimartini** [comment]
Hey @vip9,

I am not sure if I understood your issue.

I think you need, always when you create an event, set the right split to it. Then it will have a specific split.

In your codepen, you haven't set the split to the event 'Swimming lesson'. Do it as a static way:

```
events: [
      {
        start: '2020-06-13 19:30',
        end: '2020-06-13 23:00',
        title: 'Swimming lesson',
        split: 1
      }
],
```

Or dynamically (in your controller, for example):

`$events['split'] = the.id.of.splitDays;`

Note: maybe you will have some issues about the split ordering. I was using the 

`$events['split'] = the.id.of.splitDays;`

with the splitDays id being the same as the id of the split database table (that in my case is `courts` (sports)). The problem comes when you remove a court from your database, then vue-cal loses the reference to the splitDays id. This is, if you have splitDays: 1, 2, 3 and courts relating these splitDays, it will work, but if you have splitDays: 1, 3, it will lose the reference, because it needs the splitDay 2. This means the court 2 was deleted and vue-cal cannot handle with ids 1, 3 only. So if you add a new court to the database with id 4, when you reference splitDays 4 to the court 4, it will mess everything (1, 3, 4 cannot be handled by vue-cal).
Vue-cal needs the splitDays array to be in sequence.

**Comment by vip9** [comment]
The problem is not with viewing, but when creating new one.
In new one event  `onEventCreate (event: any, deleteEventFunction: any) {`;
`event` has no property "split".
Updated the codepen with latest vue-cal version.

**Comment by giulianimartini** [comment]
Maybe @antoniandre can explain why it has no split in your case.

But, a workaround is to change `:on-event-create` to `@cell-click="onEventCreate"`.

Then you have the `date` and `split` properties.

You can try to create an event ([documentation example](https://antoniandre.github.io/vue-cal/#ex--other-event-creation-methods)) by setting a `ref="vue-cal"` to your vue-cal instance and then:

```
$refs.vuecal.createEvent(
    $event,
    120,
    { title: 'New Event', class: 'blue-event' }
)
```

Hope it helps.

**Comment by antoniandre** [comment]
Thanks @giulianimartini for the help.
Thanks @vip9 for reporting, I believe this is a regression.
Will release a fix soon.

**Comment by antoniandre** [comment]
Hi @vip9,

As @giulianimartini said, `:drag-to-create-event="false"` is needed. It is not a trick: the default event creation method has changed since version 3.5.0 in favor of drag creation.

Regarding the rest of the problem, this was indeed a regression introduced with the new drag creation method. I released a fix in version 3.5.3 and your codepen example is now working.

Thanks for reporting the issue.
Please reopen if not solved.
